### ISSUE TITLE
Feature: `useBackgroundQuery` hook

### ISSUE THREAD

**Issue by hwillson** [apollographql/apollo-client#8694]
This is a placeholder issue for tracking purposes. Details in https://github.com/apollographql/apollo-client/pull/8783.

this new `useBackgroundQuery` hook allows a component to fetch data in a way that's managed by the React component lifecycle, without automatically subscribing to every future change in that data, allowing child components to update in place without requiring the parent component to rerender all its children.

in other words: `useBackgroundQuery` executes as soon as the component mounts and can be used  as a render optimization so you can ensure your component doesn't oversubscribe to data changing in the cache that it doesn't care about which can help you avoid some rerenders. The prime use case for this is lists. You might not want a parent component to rerender every time data from a single item in the list is updated.

**Comment by jpvajda** [comment]
@hwillson should this issue reference https://github.com/apollographql/apollo-client/pull/8783? it seems to be pointing to itself. 😄

**Comment by hwillson** [comment]
Ooops - fixed, thanks @jpvajda! 🙇

**Comment by alessbell** [comment]
Our new design for `useBackgroundQuery` is as a data fetching hook that's part of Apollo Client's Suspense story: it will provide a way to initiate queries in the "background" while rendering continues until a component consuming the data further down the render tree throws a promise and suspends, thus enabling the "render as you fetch" pattern.

This diverges from the original design of the hook described in this issue as one that pairs with `useFragment` to allow list items to update in place, which will be solved by a `@nonreactive` directive.

**Comment by alessbell** [comment]
This work was completed in https://github.com/apollographql/apollo-client/pull/10755 and will be released with the next 3.8 alpha 🎉
